docs/source/format/FlightSql.rst

```diff
@@ -141,6 +141,13 @@ the ``type`` should be ``ClosePreparedStatement``).
     Execute a previously created prepared statement and get the results.
 
     When used with DoPut: binds parameter values to the prepared statement.
+    The server may optionally respond with an updated handle. The client
+    should use this updated handle for all subsequent requests for this
+    prepared statement. The updated handle allows implementing query 
+    parameters with stateless services. Note that the server is responsible 
```

Comment:
This is very difficult to understand. How does an updated handle allow implementing query parameters? This should be more descriptive (does the handle embody the bound parameters?).

Also, "the client should use this updated handle" means the client cannot bind the original prepared statement to other parameters?

Reply:
> . How does an updated handle allow implementing query parameters? This should be more descriptive (does the handle embody the bound parameters?).

Maybe we can say something like this:

> Optionally updating the handle allows the client to supply all state required to execute the query in an `ActionPreparedStatementExecute` message. For example, stateless servers can encode the bound parameter values into the new handle, and the client will send that new handle with parameters back to the server. 


The problem is  as currently written, the `CommandPrepareStatementQuery` message contains parameter values, but does not return anything to the client prior to the client sending `ActionPreparedStatementExecute`. Thus the only way to implement prepared statements with bind parameters today is to store the value of the parameters on the server between the call to `CommandPrepareStatementQuery`  and `ActionPreparedStatementExecute`.


> Also, "the client should use this updated handle" means the client cannot bind the original prepared statement to other parameters?

That is an excellent question. I think the intent is that the handle could be updated with new parameter values. I agree as worded this is confusing and should be clarified 

>  Note that a handle returned from a DoPut call with `CommandPreparedStatementQuery` can itself be passed to a subsequent DoPut call with `CommandPreparedStatementQuery` to bind a new set of parameters. The subsequent call itself may return an updated handled which again should be used for subsequent requests.